contracts/DomainRegistry.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+contract DomainRegistry {
+    uint256 constant REQUIRED_DEPOSIT = 1 ether;
+
+    struct Domain {
+        uint256 deposit;
+        bool isRegistered;
+    }
+
+    mapping(string => Domain) public domains;
+
+    event DomainCreated(string tld, uint256 deposit);
+    event DomainReleased(string tld);
+
+    modifier hasRequiredDeposit(uint256 _requiredDeposit) {
+        require(msg.value >= _requiredDeposit, "Wrong eth amount");
+        _;
+    }
+
+    modifier domainDoesNotExist(string memory _tld) {
+        require(!domains[_tld].isRegistered, "Domain exists");
+        _;
+    }
+
+    modifier isTopLevelDomain(string memory _tld) {
+        require(bytes(_tld).length > 0, "Domain is empty");
+        require(bytes(_tld)[0] != bytes(".")[0], "Multilevel domains are not allowed");
+        require(bytes(_tld)[bytes(_tld).length - 1] != bytes(".")[0], "Multilevel domains are not allowed");
```

Comment:
What if point is in not at the first or last symbol in the string?

Reply:
fxd